source/content/guides/secrets/05-drupal.md

```diff
@@ -0,0 +1,88 @@
+
+- Make sure you have [terminus installed](https://docs.pantheon.io/terminus/install#install-terminus) in your machine
+
+- Install the [Secrets Manager Plugin](https://github.com/pantheon-systems/terminus-secrets-manager-plugin#installation)
+
+### Steps
+
+1. Install the required modules in your Drupal site and push the changes to Pantheon:
+    ```
+    composer require drupal/pantheon_secrets drupal/sendgrid_api drupal/sendgrid_mailer
+    git add composer.json composer.lock
+    git commit -m "Add required modules."
+    git push
+    ```
+
+1. Enable the modules:
+    ```
+    terminus drush <site>.<env> -- en -y pantheon_secrets sendgrid_api sendgrid_mailer
+    ```
+
+1. Make sure your Sendgrid account is correctly configured and allows sending email.
+
+1. Create a Sendgrid API key by following [Sendgrid instructions](https://docs.sendgrid.com/ui/account-and-settings/api-keys#creating-an-api-key)
+
+1. Store the API key as a site secret:
+    ```
+    terminus secret:site:set <site> sendgrid_api <api_key> --scope=web --type=runtime
+    ```
+
+    As a best practice, the non-production environments should be the default and then override that value with a [secret environment override](/guides/secrets/overview#environment-override) to change the API key for the live environment (e.g. you want to use different Sendgrid accounts for live and dev environments)
+
+1. Add the Key entity in one of the different available ways:
+
+    1. Add a new key through the Key module UI. Select Pantheon Secret as the key provider and your secret name from the dropdown (make sure you select "Sendgrid" as the Key type and "Pantheon" as the Key provider)
+
+        ![Screenshot of creating a new Key entity with type "Sendgrid" and provider "Pantheon"](../../../images/guides/secrets/add-key.png)
+
+    1. Go to `/admin/config/system/keys/pantheon` and click on the "Sync Keys" button to get all of the available secrets into Key entities.
+
+        ![Screenshot of Sync Pantheon Secrets page in Drupal UI](../../../images/guides/secrets/sync-keys.png)
+
+        Then, edit the `sendgrid_api` Key and change the type to "Sendgrid"
+
+    1. Use the provided drush command to sync all of your secrets into Key entities:
+        ```
+        terminus drush <site>.<env> -- pantheon-secrets:sync
+        ```
+
+        Then, edit the `sendgrid_api` Key and change the type to "Sendgrid"
+
+1. Go to the Sendgrid API Configuration page (`/admin/config/services/sendgrid`) and select your Key item
+
+    ![Screenshot of Sendgrid API Configuration page in Drupal UI](../../../images/guides/secrets/sendgrid-config.png)
+
+1. Make sure your site "Email Address" (`/admin/config/system/site-information`) matches a verified Sender Identity in Sendgrid
+
+1. Go to the Sendgrid email test page (`/admin/config/services/sendgrid/test`) and test your Sendgrid integration by sending a test email
```

Comment:
At some point before this step, don't we need to configure default mail system? 
<img width="1198" alt="Screenshot 2024-08-06 at 5 17 59 PM" src="https://github.com/user-attachments/assets/0d77f41b-ca51-418c-aef8-e057e0f3e200">

Reply:
@jazzsequence here's what made me ask: https://git.drupalcode.org/project/sendgrid_mailer#installation 

I'm having issues with my test Sendgrid account. When I update default mailsystem to sendgrid mailer I get the following [sendgrid error message](https://www.twilio.com/docs/api/errors/60604) (I think due to the status of my inactive sendgrid account I can't authorize senders but I can somehow generate API keys) 
```
{"errors":[@"message":"Authenticated user is not authorized to send mail","field":null,"help":null]}
```
If I set default mailsystem to PHP mailer, I don't get that error message but also the email doesn't show up or when it does it lands in spam hours later 

I think we need to update default mailsystem config in order for this setup to work, but I'm not in a position to easily validate that myself

Reply:
The link you added is correct for the complete configuration; in the original version of these instructions, I wasn't attempting to substitute instructions to setup sendgrid but how to set it up with secrets manager.

To test using the Sendgrid Test page, you don't need to setup mailsystem stuff, for the final site usage, probably yes

Reply:
I was able to debug my sendgrid account and test this directly. 

I didn't find this to be true: 
> To test using the Sendgrid Test page, you don't need to setup mailsystem stuff 

I found that if you don't configure Sendgrid mailer for default mail system, the test email is not sent using my 3rd party sendgrid account, it's sent from Pantheon's default PHP mail system and it's not reliably delivered.

I'll add this as a new step in the doc